test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiSchemaTests.cs

```diff
@@ -424,11 +444,64 @@ public void ParseAdvancedSchemaWithReferenceShouldSucceed()
                                             }
                                         }
                                     }
+                                },
+                                Reference = new OpenApiReference()
+                                {
+                                    Id= "Dog",
+                                    Type = ReferenceType.Schema
                                 }
                             }
                         }
                     });
             }
         }
+
+
+        [Fact]
+        public void ParseSelfReferencingSchemaShouldNotStackOverflow()
+        {
+            using (var stream = Resources.GetStream(Path.Combine(SampleFolderPath, "selfReferencingSchema.yaml")))
+            {
+                // Act
+                var openApiDoc = new OpenApiStreamReader().Read(stream, out var diagnostic);
+
+                // Assert
+                var components = openApiDoc.Components;
+
+                diagnostic.ShouldBeEquivalentTo(
+                    new OpenApiDiagnostic() { SpecificationVersion = OpenApiSpecVersion.OpenApi3_0 });
+
+                //var schemaExtension = new OpenApiSchema()
+                //{
+                //    AllOf = { new OpenApiSchema()
+                //    {
+                //        Title = "schemaExtension",
+                //        Type = "object",
+                //        Properties = {
+                //                        ["description"] = new OpenApiSchema() { Type = "string", Nullable = true},
+                //                        ["targetTypes"] = new OpenApiSchema() {
+                //                            Type = "array",
+                //                            Items = new OpenApiSchema() {
+                //                                Type = "string"
+                //                            }
+                //                        },
+                //                        ["status"] = new OpenApiSchema() { Type = "string"},
+                //                        ["owner"] = new OpenApiSchema() { Type = "string"},
+                //                        ["child"] = null
+                //                    }
+                //        }
+                //    },
+                //    Reference = new OpenApiReference()
+                //    {
+                //        Type = ReferenceType.Schema,
+                //        Id = "microsoft.graph.schemaExtension"
+                //    }
+                //};
+
+                //schemaExtension.AllOf[0].Properties["child"] = schemaExtension;
```

Comment:
Does this fail the unit test due to the loop? If so, you can set xUnit to only delve down until specified level just for this test

Reply:
I couldn't figure out why it was failing.  I kept getting an error that made no sense.  Anyway, I just tried running it again and it is working now.